### PR TITLE
MAUI-5238 : Prepare UG for Popup Events and Commands in .NET MAUI popup

### PR DESCRIPTION
**Pull request by RiyashameedM**
Prepared UG for Popup Events and Commands in .NET MAUI popup

### CODE REVIEW

File: MAUI/Popup/Popup-Events.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+{% highlight c# tabtitle="MainPage.xaml.cs" hl_lines"4" %}
+public MainPage()
+{
+    InitializeComponent();
+    popup.Opening += Popup_Opening;
+}
+
+private void Popup_Opening(object sender, System.ComponentModel.CancelEventArgs e)
+{
+    e.Cancel = true;
```
Comment (PiruthivirajM) [inline MAUI/Popup/Popup-Events.md:39]: Highlight this line too

File: maui-toc.html [hunk 2]
```diff
@@ -506,11 +506,10 @@
             <li><a href="/maui/Popup/overview">Overview</a></li>
             <li><a href="/maui/Popup/getting-started">Getting Started</a></li>
             <li><a href="/maui/Popup/migration">Migrate from Xamarin.Forms</a></li>
-
+            <li><a href="/maui/Popup/Popup-Events">Popup Events And Commands</a></li>
```
Comment (PiruthivirajM) [inline maui-toc.html:509]: Event and Commands

**Review by PiruthivirajM — COMMENTED** [review]
Address review comments

File: maui-toc.html (hunk 2 above)
Comment (RiyashameedM) [reply maui-toc.html]: modified

File: MAUI/Popup/Popup-Events.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+## Accept command
+
+The `SfPopup.AcceptCommand` will be fired when clicking the Accept button in the popup footer.
+
+To handle the Accept button,
+
+* Derive a class from `ICommand`, and implement the `ICommand` interface.
+* To prevent popup from closing, return false in the `CanExecute()` override method and the `Execute()` override method will not be fired.
+* Else return true in the `CanExecute()` override method and do the required operations in the `Execute()` method.
+* Now create a property of your custom command type in the view model class and initialize it.
```
Comment (M-A32) [inline MAUI/Popup/Popup-Events.md:128]: * Now, create a property of your custom command type in the view model class and initialize it.

File: MAUI/Popup/Popup-Events.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+{% endtabs %}
+
+## Decline command
+
+The `SfPopup.DeclineCommand` will be fired when clicking the Decline button in the popup footer. 
+
+To handle the Decline button,
+
+* Derive a class from `ICommand`, and implement the `ICommand` interface.
```
Comment (M-A32) [inline MAUI/Popup/Popup-Events.md:208]: * Derive a class from the `ICommand`, and implement its interface.

**Review by M-A32 — CHANGES_REQUESTED** [review]

File: MAUI/Popup/Popup-Events.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+   // Codes that needs to be executed once popup is visible in the screen.
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Closing event
+
+The `SfPopup.Closing` event will be fired whenever closing the Popup in the application. It can cancel popup closing with `CancelEventArgs` that contains the following property:
+
+* `Cancel`: Popup opening is based on this value.
```
[the remark this answers is not in the document]
Comment (RiyashameedM) [reply MAUI/Popup/Popup-Events.md]: resolved

File: MAUI/Popup/Popup-Events.md [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+{% endhighlight %}
+{% endtabs %}
+
+## Accept command
+
+The `SfPopup.AcceptCommand` will be fired when clicking the Accept button in the popup footer.
+
+To handle the Accept button,
+
+* Derive a class from `ICommand`, and implement the `ICommand` interface.
```
[the remark this answers is not in the document]
Comment (RiyashameedM) [reply MAUI/Popup/Popup-Events.md]: modified

File: MAUI/Popup/Popup-Events.md [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+{% endtabs %}
+
+## Decline command
+
+The `SfPopup.DeclineCommand` will be fired when clicking the Decline button in the popup footer. 
+
+To handle the Decline button,
+
+* Derive a class from `ICommand`, and implement the `ICommand` interface.
+* To prevent popup from closing, return false in the `CanExecute()` override method and the `Execute()` override method will not be fired.
```
[the remark this answers is not in the document]
Comment (RiyashameedM) [reply MAUI/Popup/Popup-Events.md]: modified

File: MAUI/Popup/Popup-Events.md [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+{% endtabs %}
+
+## Accept command
+
+The `SfPopup.AcceptCommand` will be fired when clicking the Accept button in the popup footer.
+
+To handle the Accept button,
+
+* Derive a class from `ICommand`, and implement the `ICommand` interface.
+* To prevent popup from closing, return false in the `CanExecute()` override method and the `Execute()` override method will not be fired.
```
[the remark this answers is not in the document]
Comment (RiyashameedM) [reply MAUI/Popup/Popup-Events.md]: modified

File: MAUI/Popup/Popup-Events.md [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+private void Popup_Opened(object sender, EventArgs e)
+{
+   // Codes that needs to be executed once popup is visible in the screen.
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Closing event
+
+The `SfPopup.Closing` event will be fired whenever closing the Popup in the application. It can cancel popup closing with `CancelEventArgs` that contains the following property:
```
[the remark this answers is not in the document]
Comment (RiyashameedM) [reply MAUI/Popup/Popup-Events.md]: modified

File: MAUI/Popup/Popup-Events.md [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+* Opening
+* Opened
+* Closing
+* Closed
+
+## Opening event
+
+The `SfPopup.Opening` event will be fired whenever opening the Popup in the application. It can cancel popup opening with `CancelEventArgs` that contains the following property:
+
+* `Cancel`: Popup opening is based on this value.
```
[the remark this answers is not in the document]
Comment (RiyashameedM) [reply MAUI/Popup/Popup-Events.md:23]: resolved

File: MAUI/Popup/Popup-Events.md [hunk 11]
```diff
@@ -0,0 +1,285 @@
+---
+layout: post
+title: Popup Events And Commands in .NET MAUI Popup control | Syncfusion
+description: Learn here all about Popup Events And Commands support in Syncfusion .NET MAUI Popup (SfPopup) control and more.
```
[the remark this answers is not in the document]
Comment (RiyashameedM) [reply MAUI/Popup/Popup-Events.md]: modified

**Comment by RiyashameedM** [comment]
rebuild

**Pull request closed without merging**